src/dataset-builder/DatasetBuilder.ts

```diff
@@ -543,17 +557,18 @@ export const DatasetBuilderContents = ({
           },
         }),
     ]),
-    snapshotAccessRequest.status === 'Loading'
+    snapshotAccessRequest.status === 'Loading' || snapshotAccessRequestDetails.status === 'Loading'
       ? spinnerOverlay
       : snapshotAccessRequest.status === 'Ready' &&
+        snapshotAccessRequestDetails.status === 'Ready' &&
```

Comment:
If details are gated on the request being ready, do you need to check both here?
```suggestion
      : snapshotAccessRequestDetails.status === 'Ready' &&
```

Since the contents of `snapshotAccessRequest` are no longer used inside the `&&` clause, we no longer need to refine its type based on the `status`.

Reply:
The contents of `snapshotAccessRequest` are used inside `&&` for the ID

Reply:
I think it is probably okay to gate on just details (although the loading check prevents potential screen flicker), but the type coercion needs to happen somewhere, and this seems as good as anything.